JavaScript/JavaScriptSDK.Tests/CheckinTests/Context/Session.tests.ts

```diff
@@ -120,6 +128,88 @@ class SessionContextTests extends TestClass {
         });
 
         this.testCase({
+            name: "ai_session local storage has correct structure",
+            test: () => {
+                if (window.localStorage) {
+                    // setup
+                    var actualCookieName: string;
+                    var actualCookieValue: string;
+                    var newGuidStub = sinon.stub(Microsoft.ApplicationInsights.Util, "newGuid",() => "newGuid");
+                    var getCookieStub = sinon.stub(Microsoft.ApplicationInsights.Util, "getCookie",() => "");
+                    var setCookieStub = sinon.stub(Microsoft.ApplicationInsights.Util, "setCookie",(cookieName, cookieValue) => { });
+
+                    // act
+                    var sessionManager = new Microsoft.ApplicationInsights.Context._SessionManager(null,() => { });
+                    sessionManager.update();
+                    sessionManager.backup();
+
+                    // verify
+                    Assert.ok(localStorage["ai_session"], "ai_session storage is set");
+
+                    Assert.equal(3, localStorage["ai_session"].split('|').length, "Cookie value before expiration should include user id, acq date and renew date");
+                    Assert.equal("newGuid", localStorage["ai_session"].split('|')[0], "First part of cookie value should be new user id guid");
+
+                    // cleanup
+                    getCookieStub.restore();
+                    setCookieStub.restore();
+                    newGuidStub.restore();
+                } else {
+                    // this might happen on IE when using a file:// url
+                    Assert.ok(true, "browser does not support local storage in current environment");
```

Comment:
Are you referring to the [RunTestsInBrowser.ps1](https://github.com/Microsoft/ApplicationInsights-JS/blob/master/RunTestsInBrowser.ps1)? I think it starts IISExpress and opens a browser to a `http://localhost` port. 

The reason I'm asking is that [conditional test logic](http://xunitpatterns.com/Conditional%20Test%20Logic.html) is a pretty bad [code smell](http://xunitpatterns.com/code%20smell.html). Unless we _really_ need it, I would like us to eliminate this if/else statement.